CheckBattery.c

```diff
@@ -0,0 +1,17 @@
+#include <stdio.h>
+#include <assert.h>
+#include "CheckBattery.h"
+
+int batteryIsOk(float temperature, float soc, float chargeRate) {
+	return (!BATTERY_PARAM_NOTINRANGE(temperature, TEMP_MIN, TEMP_MAX) && !BATTERY_PARAM_NOTINRANGE(soc, SOC_MIN, SOC_MAX) && 
```

Comment:
Suggest to avoid use of double negative... `!` means 'not'. The macros are also about 'NOT' INRANGE.
See how it looks when you change it to being 'in range'. Like BATTERY_PARAM_ISINRANGE